src/TimeIntegration.cpp

```diff
@@ -119,7 +121,11 @@ void TimeIntegration::run() {
 #pragma acc update host(d_nu_t[:bsize])
 #pragma acc update host(d_S_T[:bsize]) wait    // all in one update does not work!
 
-            m_visual->visualise(m_field_controller, t_cur);
+            if (m_has_analytical_solution) {
+                m_solution->calc_analytical_solution(t_cur);
+            }
```

Comment:
this inquiry does not belong to the TimeIntegration class. It can be in the Visual class as before or in the Solution class itself. Either way, this shouldn't be part of this PR.

Reply:
since `Solution::calc_analytical_solution` is not const but `Visual::visualise` is now we had to move it. The only option left is `Solution`. Done in e42733f